packages/api/cms-api/src/page-tree/site-preview.resolver.ts

```diff
@@ -0,0 +1,36 @@
+import { Inject } from "@nestjs/common";
+import { Args, Query, Resolver } from "@nestjs/graphql";
+import { GraphQLJSONObject } from "graphql-scalars";
+import { SignJWT } from "jose";
+
+import { RequiredPermission } from "../user-permissions/decorators/required-permission.decorator";
+import { ContentScope } from "../user-permissions/interfaces/content-scope.interface";
+import { SITE_PREVIEW_CONFIG } from "./page-tree.constants";
+
+export type SitePreviewConfig = {
+    secret: string;
+};
+
+@Resolver()
+export class SitePreviewResolver {
+    constructor(@Inject(SITE_PREVIEW_CONFIG) private readonly config: SitePreviewConfig) {}
+
+    @Query(() => String)
+    @RequiredPermission("pageTree")
+    async sitePreviewJwt(
+        @Args("scope", { type: () => GraphQLJSONObject }) scope: ContentScope,
+        @Args("path") path: string,
+        @Args("includeInvisible") includeInvisible: boolean,
```

Comment:
Shouldn't we use the x-include-invisible-content header for this?

Reply:
What happens when using the button in the site preview? Will this generate a new JWT?